src/relay.cpp

```diff
@@ -800,10 +815,11 @@ void server_callback(evutil_socket_t fd, short event, void *arg) {
     sockaddr_in6 from;
     socklen_t len = sizeof(from);
     int32_t data = 0;
-    static uint8_t message_buffer[4096];
+    static uint8_t message_buffer[BUFFER_SIZE];
 
-    if ((data = recvfrom(config->local_sock, message_buffer, 4096, 0, (sockaddr *)&from, &len)) == -1) {
-        syslog(LOG_WARNING, "recv: Failed to receive data from server\n");
+    if ((data = recvfrom(config->local_sock, message_buffer, BUFFER_SIZE, 0, (sockaddr *)&from, &len)) == -1) {
+        syslog(LOG_ERR, "recv: Failed to receive data from server\n");
+        return;
     }
 
```

Comment:
We may need add protection code to make sure message_buffer length large than sizeof struct dhcpv6_msg( ).

as we referred to  msg->msg_type immediately.